.github/workflows/build_publish.yml

```diff
@@ -0,0 +1,63 @@
+name: Build & Publish
+
+on:
+  pull_request:
+    branches: [master]
+  push:
+
+jobs:
+  Build:
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      # Install Deps
+      - uses: lukka/get-cmake@latest
+
+      # MSVC
+      - name: Set up Visual Studio shell
+        uses: egor-tensin/vs-shell@v2
+        with:
+          arch: Win32
+
+      - name: Build w/ MSVC
+        run: ninja -f build\msvc.ninja
+
+      # LLVM
+      - name: Cache LLVM and Clang
+        id: cache-llvm
+        uses: actions/cache@v2
+        with:
+          path: |
+            C:/Program Files/LLVM
+            ./llvm
+          key: llvm-14
+
+      - name: Install LLVM and Clang
+        uses: KyleMayes/install-llvm-action@v1
+        with:
+          version: "14.0.6"
+          cached: ${{ steps.cache-llvm.outputs.cache-hit }}
+
+      - name: Build w/ LLVM
+        run: ninja -f build\llvm.ninja
+
+      # GCC
+      - name: Set up MinGW
+        uses: egor-tensin/setup-mingw@v2
+        with:
+          platform: i686
```

Comment:
Mingw-w64 exists in the default windows agent, but the 64 version annoyingly won't compile the 32-bit binary no matter what i tried, so we reinstall it proper